packages/core/src/provider.ts

```diff
@@ -350,7 +350,7 @@ export async function getBalance(
     const balanceHex = await provider.request({ method: 'eth_getBalance', params:[address,'latest'] })
     const balanceWei = new BigNumber(parseInt(balanceHex, 16))
     return balanceWei
-      ? { [chain.token || 'eth']: weiToEth(balanceWei) }
+      ? { [chain.token || 'eth']: weiToEth(balanceWei.toString()) }
```

Comment:
Since we are passing in a string, we could just pass in the `balanceHex` value directly to the `weiToEth` function.

Reply:
Yeah. Figured out introducing the BigNumber was an overkill. Thanks for pointing it out.